syntax = "proto3";

package grpcfs;

// Interface exported by the server.
service FileServer {
  rpc GetFile(FileRequest) returns (FileRequestAnswer) {}
}

message FileRequest {
  string name = 1;
  string host = 2;
  int32 port = 3;
}

message FileRequestAnswer {
  enum Status {
    OK = 0;
    FAIL = 1;
    FULL = 2;
  }
  Status status = 1;
}